dataactcore/utils/fileF.py

```diff
@@ -159,18 +153,27 @@ def transform(self, models):
 
 # Collect the models associated with a single F CSV row
 ModelRow = namedtuple(
-    'ModelRow', ['award', 'procurement', 'subcontract', 'grant', 'subgrant'])
+    'ModelRow',
+    ['award', 'procurement', 'subcontract', 'grant', 'subgrant', 'naics_desc'])
+ModelRow.__new__.__defaults__ = (None, None, None, None, None)
 
 
 def submission_procurements(submission_id):
     """Fetch procurements and subcontracts"""
-    triplets = GlobalDB.db().session.\
-        query(AwardFinancial, FSRSProcurement, FSRSSubcontract).\
+    sess = GlobalDB.db().session
+
+    naics_subquery = sess.query(AwardProcurement.naics_description).\
+        filter(AwardProcurement.submission_id == submission_id,
+               AwardProcurement.piid == AwardFinancial.piid,
+               AwardProcurement.naics == FSRSSubcontract.naics).\
+        order_by(AwardProcurement.award_procurement_id).limit(1).as_scalar()
+    results = sess.query(AwardFinancial, FSRSProcurement, FSRSSubcontract,
+                         naics_subquery).\
         filter(AwardFinancial.submission_id == submission_id).\
         filter(FSRSProcurement.contract_number == AwardFinancial.piid).\
         filter(FSRSSubcontract.parent_id == FSRSProcurement.id)
-    for award, proc, sub in triplets:
-        yield ModelRow(award, proc, sub, None, None)
+    for award, proc, sub, naics_desc in results:
```

Comment:
Why are we combining the filter in the query above and not in this one? Seems like we should pick one method and stick to it (at least when the two filters are so close to one another)